zebra-consensus/src/primitives.rs

```diff
@@ -1,5 +1,9 @@
 //! Asynchronous verification of cryptographic primitives.
```

Comment:
This PR is waiting for a full sync before it merges.

Reply:
This mainnet sync:
https://github.com/ZcashFoundation/zebra/actions/runs/5490276483

And I just started a testnet sync here:
https://github.com/ZcashFoundation/zebra/actions/runs/5511409905

Reply:
I accidentally cancelled the testnet sync when I was trying to cancel just the checkpoint and lightwalletd jobs. I'm really sorry about that!

I started another testnet sync here:
https://github.com/ZcashFoundation/zebra/actions/runs/5514327024

The checkpoint and lightwalletd jobs will run because they don't have cached states for testnet. If they fail, we can ignore them and open a PR to skip them in manual testnet runs.

Reply:
> This mainnet sync:
> https://github.com/ZcashFoundation/zebra/actions/runs/5490276483

This one passed.

Reply:
The mainnet full sync took 182232s (50.6 hours):
https://github.com/ZcashFoundation/zebra/actions/runs/5490276483/jobs/10071236412#step:8:19629

Compared to 179831s (50.0h) and 180705s (50.2h) for the last two scheduled syncs (Friday's CI failed due to a temporary network error, but the test actually finished):
https://github.com/ZcashFoundation/zebra/actions/runs/5422688804/jobs/9886836808#step:8:17919
https://github.com/ZcashFoundation/zebra/actions/runs/5486195320/jobs/10023005486#step:8:18014

That's a 1% performance decrease, which is ok, it's about what we'd expect over 3 weeks, and is partly due to the full verification since the last checkpoint update.

But it also produces 1600 more log lines (30 per hour). Would you mind checking that out? It doesn't really matter much, but I thought it was worth pointing out.

Reply:
The extra log lines might have been a log we deleted in the last 135 commits. The branch we were running the manual sync from hadn't been rebased in a while.

Reply:
> The extra log lines might have been a log we deleted in the last 135 commits.

Yeah, it may have been missing a change from `info` level logging to `debug` in `route_inv()`, I don't think there would be 1600 extra handshake errors if `reconnection_peers()` checks by IP but perhaps a few there.

Reply:
It's likely either overload logs:
> 2023-07-10T22:49:59.623623Z  INFO {net="Main"}:{peer=Out("v4redacted:8233")}:msg_as_req{msg="inv"}: zebra_network::peer::connection: inbound service is overloaded, closing connection drop_connection_probability=0.05000000074505806 remote_user_agent="/MagicBean:5.6.1/" negotiated_version=Version(170100) peer="v4redacted:8233" last_peer_state=Some("AwaitingRequest::In::Req::AdvertiseTransactionIds") remote_height=Height(2151066) cached_addrs=2 connection_state=AwaitingRequest

(There are at least a few hundred of these.)

The duplicate IP address info messages:
> 2023-07-11T13:10:47.276913Z  INFO {net="Main"}:crawl_and_dial{new_peer_interval=30s}: zebra_network::peer_set::set: duplicate IP addresses in peer set duplicate_connections=2 duplicated_peers=1 peers=75

Or various logs or warnings that don't happen since we fixed the 100% CPU future.